src/odemis/driver/orsay.py

```diff
@@ -2886,68 +2878,58 @@ def on_connect(self):
-                                                          recursive_getattr(self._hybridAperture,
-                                                                            f"Aperture{aprtr_nmbr}.PositionX"))
-                                  )
-            connector_refs.append(OrsayParameterConnector(self._apertureDict[f"Aperture{aprtr_nmbr}"]["PositionY"],
-                                                          recursive_getattr(self._hybridAperture,
-                                                                            f"Aperture{aprtr_nmbr}.PositionY"))
-                                  )
-
         self.update_VAs()
-        self._metadata[model.MD_APERTURES_INFO] = self._getAllApertureMetadata()
 
-    def selectAperture(self, id):
-        return self._executor.submit(self._selectAperture, id)
+    def connectApertureDict(self):
+        """
+        (re)Connects the aperture dict to the Orsay server by subscribing listeners to all the aperture data and
+        updating the values.
+        """
+        # Remove the references to the old connectors.
+        self._apertureConnectors = {i: [] for i in range(self._nmbr_apertures)}
 
-    def _selectAperture(self, id, timeout=10):
-        logging.debug(f"Aperture {id} is selected")
-        tend = time.time() + timeout
-        self._hybridAperture.SelectedDiaph.Target = id
-        while int(self._hybridAperture.SelectedDiaph.Actual) != id \
-                and self._hybridAperture.IsMovingX.Actual == "True" \
-                and self._hybridAperture.IsMovingY.Actual == "True":
-            time.sleep(0.1)
-            t = time.time()
-            if t > tend:
-                raise TimeoutError("Referencing timeout after %g s" % timeout)
+        for aprtr_nmbr in range(self._nmbr_apertures):
+            # Subscribe to and update the value Lifetime
+            Orsay_parameter = recursive_getattr(self._hybridAperture, f"Aperture{aprtr_nmbr}.Lifetime")
+            self._addApertureListener(aprtr_nmbr, "Lifetime", Orsay_parameter)
+            self._apertureDict[aprtr_nmbr]["LifeTime"] = self._apertureVarTypes["LifeTime"](Orsay_parameter.Actual)
+            
+            # Subscribe to and update the value Size
+            Orsay_parameter = recursive_getattr(self._hybridAperture, f"Aperture{aprtr_nmbr}.Size")
+            self._addApertureListener(aprtr_nmbr, "Size", Orsay_parameter)
+            self._apertureDict[aprtr_nmbr]["Size"] = self._apertureVarTypes["Size"](Orsay_parameter.Actual)
 
-        self._selectedApertureConnector.update_VA()
-        self._sizeSelectedApertureConnector.update_VA()
+            # Subscribe to and update the value PositionX
+            Orsay_parameter = recursive_getattr(self._hybridAperture, f"Aperture{aprtr_nmbr}.PositionX")
+            self._addApertureListener(aprtr_nmbr, "PositionX", Orsay_parameter)
+            self._apertureDict[aprtr_nmbr]["Position"]["x"] = self._apertureVarTypes["PositionX"](Orsay_parameter.Actual)
```

Comment:
I did try that but because the structure of the dict is different than that of the Orsay Parameter this actually became less readable. I choose to rather have these lines which might seems duplicated but are better to read than multiple if statements in `addApertureListener`. The latter resulted in less readable code (to my opinion)

Reply:
Ok maybe I don't fully understand your comment but I don't call the listener. That would also be hard since I need to have the updated value to do so. Therefore I get the current value, convert that from a string to the correct type and put it in the dict.
Do you suggest to include that behavior in the listener?